examples/BERT/ns_task.py

```diff
@@ -0,0 +1,251 @@
+import argparse
+import time
+import math
+import torch
+import torch.nn as nn
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.data import DataLoader
+from model import NextSentenceTask
+from utils import run_demo, run_ddp, wrap_up
+
+
+def process_raw_data(whole_data, args):
+    processed_data = []
+    for _idx in range(len(whole_data)):
+        item = whole_data[_idx]
+        if isinstance(item, list):
+            item = torch.tensor(item)
+        if len(item) > 1:
+            # idx to split the text into two sentencd
+            split_idx = torch.randint(1, len(item), size=(1, 1)).item()
+            # Index 2 means same sentence label. Initial true int(1)
+            processed_data.append([item[:split_idx], item[split_idx:], 1])
+    # Random shuffle data to have args.frac_ns next sentence set up
+    shuffle_idx1 = torch.randperm(len(processed_data))
+    shuffle_idx2 = torch.randperm(len(processed_data))
+    num_shuffle = int(len(processed_data) * args.frac_ns)
+    shuffle_zip = list(zip(shuffle_idx1, shuffle_idx2))[:num_shuffle]
+    for (i, j) in shuffle_zip:
+        processed_data[i][1] = processed_data[j][0]
+        processed_data[i][2] = int(0)  # Switch same sentence label to false 0
+    return processed_data
+
+
+def collate_batch(batch, args, cls_id, sep_id, pad_id):
```

Comment:
You have "collate_batch" earlier as well. Is there some combination possible here with a utils file?